tests/testthat/test_ship_reports.R

```diff
@@ -0,0 +1,65 @@
+context("Test ship reports")
+
+odir <- getwd()
+on.exit(setwd(odir))
+
+setwd(tempdir())
+random_factory(include_examples = TRUE)
+source_name <- "foo"
+report_source_file_name <- list_reports(pattern = source_name)[1]
+
+dots_args <- list("lots" = data.frame(a = c(10,20)))
+compile_report(
+  report_source_file_name, 
+  quiet = FALSE, 
+  params = list("other" = "two",
+                "more" = list("thing" = "foo")),
+  extra = dots_args)
+
+compile_report(
+  report_source_file_name,
+  quiet = TRUE,
+  params = list(other = "test"))
+
+compile_report(
+  report_source_file_name, 
+  quiet = FALSE, 
+  params = list("other" = "two",
+                "more" = list("thing" = "foo")),
+  extra = dots_args)
+
+log_file <- readRDS(".compile_log.rds")
+length(log_file)
+
+test_that("Report outputs are copied into a folder at factory root", {
+  skip_on_cran()
+  
+  # entries missing some params arguments so should return an empty list
+  ship_reports(
+    params = list("other" = "two"), 
+    most_recent = TRUE
+  )
```

Comment:
since this doesn't copy the file twice, going to leave this as is for now for the sake of shipping it (the PR has gotten too big)

Reply:
I think that's fair. The only thing I'm worried about is the fact that it flattens the directory structure, which might affect the html output if it has relative paths to the figures.

Reply:
Yep, it will. My hope is people use self-contained html documents by default. We can always file that as a separate issue later, but I think that should not be a deal-breaker for the first version of shipping.

Reply:
this will be fixed in an immediate follow up PR

Reply:
+1 to fixing later; there will be questions of how we handle cases where several folders with identical names (e.g. `/figures/`) are shipped from different output folders, but I guess this is already an issue with file names as it is.